src/wallet/wallet2.cpp

```diff
@@ -7577,6 +7581,8 @@ bool wallet2::unlock_keys_file()
 
 bool wallet2::is_keys_file_locked() const
 {
+  if (m_wallet_file.empty())
+    return false;
```

Comment:
```suggestion
    return true;
```

Reply:
Why ?

Reply:
Cause when i lock the file, i expect it to be locked.

Actually would be correct to introduce a bool storing the lock state.

Reply:
OTOH, if you unlock it, you expect it to be unlocked :) Claiming locked all the time would prevent the wallet starting. Adding state would also work though.

Reply:
Given that it doesn't have state right now, it has to return `true` at least.
`false` will result in throwing the following exception:

https://github.com/monero-project/monero/blob/4b38400336c2bd5b0d011da980ca9e46862a50b9/src/wallet/wallet2.cpp#L5426

Reply:
How can you get this ? The patch passed the functional tests, which use memory wallets (and were trying to lock ".keys").

Reply:
Not sure there is a way to trigger the condition manually or through RPC interface.

Reply:
This is only called if there is a file to load from AFAICT.